.circleci/config.yml

```diff
@@ -102,6 +126,24 @@ jobs:
           name: Run cargo test
           command: cargo test --all-features
 
+  publish:
+    parameters:
+      rust_channel:
+        type: enum
+        enum: ["stable"]
+        default: stable
+      platform:
+        type: executor
+    executor: << parameters.platform >>
+    steps:
+      - checkout
+      - install_system_deps:
+          rust_channel: << parameters.rust_channel >>
+          platform: << parameters.platform >>
+      - run:
```

Comment:
Presumably we'd need to run `cargo login -- $CRATES_IO_API_TOKEN` before doing publish, while putting `CRATES_IO_API_TOKEN` into some new CircleCI context (or if there's already some CircleCI context that the `apollographql` org uses for publishing, we could use that). Or if Rust folks are doing something else to auth with `crates.io`, we can use that.

Reply:
I was wondering about this -> I based this on federation-rs where I dont see any explicit login setup. I'm guessing it is already configured on circle ci.